In [ ]:
from sklearn import datasets

In [ ]:
data=datasets.load_boston()

In [ ]:
data.DESCR


In [ ]:
import numpy as np
import pandas as pd

In [ ]:
df = pd.DataFrame(data.data,columns=data.feature_names)
target = pd.DataFrame(data.target, columns=["MEDV"])

OLS - ordinary least square

In [14]:
import statsmodels.api as  sm
X= df["RM"]
y= target["MEDV"]
model = sm.OLS(y,X).fit()
predictions =  model.predict(X)
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                   MEDV   R-squared:                       0.901
Model:                            OLS   Adj. R-squared:                  0.901
Method:                 Least Squares   F-statistic:                     4615.
Date:                Mon, 04 Jun 2018   Prob (F-statistic):          3.74e-256
Time:                        00:29:23   Log-Likelihood:                -1747.1
No. Observations:                 506   AIC:                             3496.
Df Residuals:                     505   BIC:                             3500.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
RM             3.6534      0.054     67.930      0.000       3.548       3.759
==============================================================================
Omnibus:                       83.295   Durbin-Watson:                   0.493
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              152.507
Skew:                           0.955   Prob(JB):                     7.65e-34
Kurtosis:                       4.894   Cond. No.                         1.00
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

coeff - when RM increases by 1 , MEDV increases by 3.65
